src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java

```diff
@@ -1167,4 +1150,79 @@ public static String calculateHMAC(final String key, final String dataToSign) {
             throw new IllegalArgumentException();
         }
     }
-}
\ No newline at end of file
+
+    public boolean isSessionValid(HttpServletRequest request) {
+        try {
+        Map<String, String> currentSessionInformation = this.getSegueSessionFromRequest(request);
+        long currentUserId = Long.parseLong(currentSessionInformation.get(SESSION_USER_ID));
+        RegisteredUser userToReturn =  database.getById(currentUserId);
+        if (null == userToReturn || !this.isValidUsersSession(currentSessionInformation, userToReturn)) {
+            log.debug("User session has failed validation. Treating as logged out. Session: " + currentSessionInformation);
+            return false;
+        }
+        return true;
+        } catch (InvalidSessionException e) {
```

Comment:
is there a reason why these exceptions can't be caught in the same catch block?

catch(InvalidSessionException | SegueDatabaseException | IOException e)